pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -739,3 +738,28 @@ func deleteJob(cj *batchv1.CronJob, job *batchv1.Job, jc jobControlInterface, re
 func getRef(object runtime.Object) (*corev1.ObjectReference, error) {
 	return ref.GetReference(scheme.Scheme, object)
 }
+
+func formatSchedule(cj *batchv1.CronJob, recorder record.EventRecorder) string {
+	timeZoneEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CronJobTimeZone)
+	scheduleContainsTZ := false
+
+	if strings.Contains(cj.Spec.Schedule, "TZ") {
+		scheduleContainsTZ = true
+		recorder.Eventf(cj, corev1.EventTypeWarning, "UnsupportedSchedule", "CRON_TZ or TZ used in schedule %q is not officially supported, see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/ for more details", cj.Spec.Schedule)
+	}
+
+	if timeZoneEnabled && cj.Spec.TimeZone != nil && len(*cj.Spec.TimeZone) > 0 {
+		if scheduleContainsTZ {
+			recorder.Eventf(cj, corev1.EventTypeWarning, "UnsupportedSchedule", "CRON_TZ and TZ cannot be used in schedule when timeZone is specified")
```

Comment:
this case is prevented by API validation, we're already recording an `UnsupportedSchedule` event above, and we're actually returning `cj.Spec.Schedule` so this message is inaccurate (we *are* honoring the CRON_TZ / TZ in the schedule)

it would be simpler to `return cj.Spec.Schedule` in the `if strings.Contains(cj.Spec.Schedule, "TZ") {` block above and drop this block